dart/test/transport/http_transport_test.dart

```diff
@@ -92,7 +92,7 @@ void main() {
           SentryEnvelope.fromEvent(sentryEvent, fixture.options.sdk);
       final eventId = await sut.send(envelope);
 
-      expect(eventId, SentryId.empty());
```

Comment:
SentryId.empty() is a `static final `, why is it failing if not `toString`?

Reply:
This got me wondering and actually `SentryId` is buggy.
The following tests do not pass:
```dart
test('id roundtrip', () {
    final id = SentryId.newId();
    expect(id, SentryId.fromId(id.toString()));
});

test('empty id equals from empty id', () {
    expect(
      SentryId.empty(),
      SentryId.fromId('00000000000000000000000000000000'),
    );
});
```
This is because it doesn't override the equality operator/hashCode.

The line you commented on failed because the second case in my example above was hit.
By calling `toString()` it passed because the string equality check then returns that they're equal.

Reply:
mm, so it's fine that it's not the same object but with the same id, but the `SentryId.empty()` is a static final var so it should be always the same, if compared with another `SentryId.empty()`

Reply:
I don't think we need to override the equality operators either, it'd be useful for tests, but for the SDK itself, it won't make any difference.

Reply:
yep, but it was converted to JSON and back to a SentryId, so that is why.